### PR TITLE
Adjust tests in array folder for new string option

### PR DESCRIPTION
**Pull request by phofl**
sits on top of #56187

### CODE REVIEW

**Comment by phofl** [comment]
cc @mroeschke gentle ping

File: pandas/core/algorithms.py [hunk 1]
```diff
@@ -929,10 +931,13 @@ def value_counts_internal(
 
             # For backwards compatibility, we let Index do its normal type
             #  inference, _except_ for if if infers from object to bool.
-            idx = Index(keys)
+            idx = Index(keys, dtype=input_dtype if input_dtype != "float16" else None)
```
Comment (mroeschke) [inline pandas/core/algorithms.py:934]: Should these changes not be here after your comments in https://github.com/pandas-dev/pandas/pull/56187?

Comment (phofl) [reply pandas/core/algorithms.py]: oh very good point, forgot the revert. thx. does

File: pandas/core/algorithms.py [hunk 2]
```diff
@@ -932,7 +932,10 @@ def value_counts_internal(
             idx = Index(keys)
             if idx.dtype == bool and keys.dtype == object:
                 idx = idx.astype(object)
-            elif idx.dtype != keys.dtype:
+            elif (
+                idx.dtype != keys.dtype  # noqa: PLR1714  # # pylint: disable=R1714
```
Comment (mroeschke) [inline pandas/core/algorithms.py:936]: Just confirming this change isn't related to your referenced, closed PR right?

Comment (phofl) [reply pandas/core/algorithms.py:936]: No this is fine, this is the infer_string inference, which will still happen

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**